exercises/strain/example.scala

```diff
@@ -0,0 +1,7 @@
+object Strain {
+  def keep[A](traversable: Traversable[A], func: A => Boolean): Traversable[A]
+    = for (a <- traversable if func(a)) yield a
```

Comment:
I agree with @abo64, this does seem a bit like cheating :) What I think the exercise leads you to is to use a plain loop for imperative languages and recursion for functional languages.